inc/saihostif.h

```diff
@@ -875,10 +875,90 @@ typedef enum _sai_hostif_table_entry_channel_type_t
     SAI_HOSTIF_TABLE_ENTRY_CHANNEL_TYPE_NETDEV_LOGICAL_PORT,
 
     /** Receive packets via Linux netdev L3 interface */
-    SAI_HOSTIF_TABLE_ENTRY_CHANNEL_TYPE_NETDEV_L3
+    SAI_HOSTIF_TABLE_ENTRY_CHANNEL_TYPE_NETDEV_L3,
+
+    /** Receive packets via Linux netdev tunnel interface */
+    SAI_HOSTIF_TABLE_ENTRY_CHANNEL_TYPE_NETDEV_TUNNEL
 
 } sai_hostif_table_entry_channel_type_t;
 
+/**
+ * @brief Attribute data for SAI_HOSTIF_TABLE_ENTRY_ATTR_CHANNEL_ENCAP_TYPE
+ */
+typedef enum _sai_hostif_channel_encap_type_t
+{
+    /** Receive packets via GRE/ERSPAN Type 2*/
+    SAI_HOSTIF_TABLE_CHANNEL_ENCAP_TYPE_ERSPAN_TYPE_II,
+
+    /** Receive packets via UDP */
+    SAI_HOSTIF_TABLE_CHANNEL_ENCAP_TYPE_UDP
+
+} sai_hostif_table_entry_channel_encap_type_t;
+
+/**
+ * @brief hostif tunnel attributes
+ */
+typedef enum _sai_hostif_tunnel_attr_t
+{
+
+    /**
+     * @brief Start of Attributes
+     */
+    SAI_HOSTIF_TUNNEL_ATTR_START,
+
+    /**
+     * @brief Tunnel source IP
+     *
+     * @type sai_ip_address_t
+     * @flags MANDATORY_ON_CREATE | CREATE_AND_SET
+     */
+    SAI_HOSTIF_TUNNEL_ATTR_SRC_IP_ADDRESS,
+
+    /**
+     * @brief Tunnel destination IP
+     *
+     * @type sai_ip_address_t
+     * @flags MANDATORY_ON_CREATE | CREATE_AND_SET
+     */
+    SAI_HOSTIF_TUNNEL_ATTR_DST_IP_ADDRESS,
+
+    /**
+     * @brief Tunnel source port
+     *
+     * @type sai_uint32_t
+     * @flags MANDATORY_ON_CREATE | CREATE_AND_SET
+     */
+    SAI_HOSTIF_TUNNEL_ATTR_SRC_PORT,
+
+    /**
+     * @brief Tunnel destination port
+     *
+     * @type sai_uint32_t
+     * @flags MANDATORY_ON_CREATE | CREATE_AND_SET
+     */
+    SAI_HOSTIF_TUNNEL_ATTR_DST_PORT,
+
+    /**
+     * @brief meta header valid
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default false
+     */
+    SAI_HOSTIF_TUNNEL_ATTR_META_HEADER_VALID,
+
+    /**
+     * @brief End of Attributes
+     */
+    SAI_HOSTIF_TUNNEL_ATTR_END,
+
+    /** Custom range base value */
+    SAI_HOSTIF_TUNNEL_ATTR_CUSTOM_RANGE_START = 0x10000000,
+
+    /** End of custom range base */
+    SAI_HOSTIF_TUNNEL_ATTR_CUSTOM_RANGE_END
+} sai_hostif_tunnel_attr_t;
```

Comment:
Where is this used?

Reply:
The primary use case for this will be sFlow.

Reply:
Why not leverage the sFlow proposal for this. In this looks like you will do the sFlow from the CPU path? I think better is to just enable sFlow, like we do from Port, from tunnel id itself?

Reply:
I removed the use of tunnel since it's being misunderstood with the tunnel infra..